backend/src/main/java/ch/puzzle/pcts/service/business/CertificateCalculationBusinessService.java

```diff
@@ -0,0 +1,127 @@
+package ch.puzzle.pcts.service.business;
+
+import static ch.puzzle.pcts.Constants.CERTIFICATE_CALCULATION;
+
+import ch.puzzle.pcts.dto.error.ErrorKey;
+import ch.puzzle.pcts.dto.error.FieldKey;
+import ch.puzzle.pcts.dto.error.GenericErrorDto;
+import ch.puzzle.pcts.exception.PCTSException;
+import ch.puzzle.pcts.model.calculation.Calculation;
+import ch.puzzle.pcts.model.calculation.certificatecalculation.CertificateCalculation;
+import ch.puzzle.pcts.model.certificatetype.CertificateKind;
+import ch.puzzle.pcts.service.persistence.CertificateCalculationPersistenceService;
+import ch.puzzle.pcts.service.validation.CertificateCalculationValidationService;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CertificateCalculationBusinessService extends BusinessBase<CertificateCalculation> {
+    private final CertificateCalculationPersistenceService certificateCalculationPersistenceService;
+    private final CertificateCalculationValidationService certificateCalculationValidationService;
+
+    protected CertificateCalculationBusinessService(CertificateCalculationValidationService validationService,
+                                                    CertificateCalculationPersistenceService persistenceService) {
+        super(validationService, persistenceService);
+        this.certificateCalculationPersistenceService = persistenceService;
+        this.certificateCalculationValidationService = validationService;
+    }
+
+    public List<CertificateCalculation> getByCalculationId(Long calculationId) {
+        return this.certificateCalculationPersistenceService.getByCalculationId(calculationId);
+    }
+
+    public List<CertificateCalculation> getByCertificateId(Long certificateId) {
+        return this.certificateCalculationPersistenceService.getByCertificateId(certificateId);
+    }
+
+    public BigDecimal getCertificatePoints(Long id) {
+        List<CertificateCalculation> certificateCalculations = this.getByCalculationId(id);
+        return certificateCalculations
+                .stream()
+                .filter(certificateCalculation -> certificateCalculation.getCalculation().getRole().getIsManagement()
+                                                  || certificateCalculation
+                                                          .getCertificate()
+                                                          .getCertificateType()
+                                                          .getCertificateKind()
+                                                     == CertificateKind.CERTIFICATE)
```

Comment:
use certificateKind.isLeaderShipexperience() instead of == CertificateKind.Certificate
this will do the same but is more concise imo